tests/api/test_authenticator.py

```diff
@@ -1639,10 +1640,18 @@ def keys():
         assert not hasattr(auth, "private_key")
         assert (public, private) == auth.key_pair
 
+    def test_key_pair_per_library(self, authenticator_fixture: AuthenticatorFixture):
         # Each library has its own key pair.
+        db = authenticator_fixture.db
+        library1 = db.default_library()
         library2 = db.library()
-        auth2 = LibraryAuthenticator.from_config(db.session, library2)
-        assert auth.public_key != auth2.public_key
+        with patch.object(Configuration, "key_pair") as patched:
+            patched.return_value = ("public", "private")
+            LibraryAuthenticator.from_config(db.session, library1)
+            assert patched.call_count == 1
+            LibraryAuthenticator.from_config(db.session, library2)
+            assert patched.call_count == 2
+            assert patched.call_args_list[0] != patched.call_args_list[1]
```

Comment:
Needed a different test to check if libraries get separate keys, since our mock always returns the same public_key.

Reply:
Minor: I think it would be useful to add a comment to the code here. Something like what you said in the comment above. Or perhaps just update the existing comment on l. 1644 to something like: "Ensure that each library obtains its own key pair."

Reply:
Added some more comments here.